megamek/src/megamek/server/ServerHelper.java

```diff
@@ -449,4 +441,510 @@ public static void checkAndApplyMagmaCrust(IHex hex, int elevation, Entity entit
             }
         }
     }
+    
+    /**
+     * Returns a list of path names of available boards of the size set in the given
+     * mapSettings. The path names are minus the '.board' extension and relative to
+     * the boards data directory.
+     */
+    static ArrayList<String> scanForBoards(MapSettings mapSettings) {
```

Comment:
In general, I'm kind of concerned that this file will become another disorganized mess like Server.java. I think it'd be better to organize the various methods into different ServerHelper classes, e.g. the board scanning/loading related methods into ServerBoardHelper, etc.

Reply:
Yes I hijacked this as my Eclipse continually goes on 10s hibernations when I edit anything in Server. I'll see if I can divide some things out.